app/grandchallenge/algorithms/tasks.py

```diff
@@ -95,26 +36,19 @@ def execute_algorithm_job_for_inputs(*, job_pk):
         kwargs={"job_pk": str(job.pk)}, immutable=True
     )
 
-    # check if all ComponentInterfaceValue's have a value.
-    missing_inputs = list(civ for civ in job.inputs.all() if not civ.has_value)
-
-    if missing_inputs:
-        job.update_status(
-            status=job.CANCELLED,
-            error_message=(
-                f"Job can't be started, input is missing for "
-                f"{oxford_comma([c.interface.title for c in missing_inputs])}"
-            ),
-        )
-        on_commit(linked_task.apply_async)
-    else:
+    if job.inputs_complete:
```

Comment:
Do we need to lock the job?

Reply:
Yes, we can lock the job but we also need to retry the task on OperationalError then since this task will be called multiple times in the process of CIV creation.